### PR TITLE
Delete comment

### PR DESCRIPTION
**Pull request by JoeDGit**

### CODE REVIEW

File: src/components/CommentCard.jsx [hunk 1]
```diff
@@ -23,6 +42,15 @@ export default function CommentCard({ comment }) {
             Posted on:{' '}
             <span style={{ fontWeight: 'bold' }}>{readableDate}</span>
           </div>
+          {author === user.username && !confirmDelete ? (
+            <button onClick={() => setConfirmDelete(true)}>Delete</button>
+          ) : null}
+          {confirmDelete ? (
+            <div>
+              Are you sure? <button onClick={handleDelete}>Yes</button> /{' '}
+              <button onClick={() => setConfirmDelete(false)}>No</button>
```
Comment (popatre) [inline src/components/CommentCard.jsx]: As an extra thing, could look at modals to do this. One possible package - https://www.npmjs.com/package/react-modal

File: src/components/CommentCard.jsx [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+  const handleDelete = () => {
+    setComments((prevComments) => {
+      return prevComments.filter((comment) => {
+        return comment.comment_id !== comment_id;
+      });
+    });
+    deleteComment(comment_id).catch(() => {
+      setComments((prevComments) => {
+        return [...prevComments, comment];
+      });
```
Comment (popatre) [inline src/components/CommentCard.jsx]: UX wise- this is confusing. 
It doesnt delete it, but instead just adds it to the bottom of the list. 
If im at the top, it looks like its gone, but actually its not deleted and is now at the bottom. 

Also, we'd ideally need some feedback to let us know its no been deleted successfully.

**Review by popatre — CHANGES_REQUESTED** [review]
Please see my comment regarding the UX of the delete

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Nice refactoring Joe, and the user gets a visual feedback that the comment has been deleted. 

One thing I noticed is, that the order of the comments is not very clear. January 2020 appears at the bottom of the list and then again close to the top too. (Learn HTML... article).

The newly posted comment also appears at the bottom of the list after the page loads in the the browser for a second time. 
I suggest to display the comments ordered by date and reversed, so the new ones end up at the top and the newly posted comment stays at the very top even after reopening/reloading the same page.

Please consider making these changes, then feel free to merge. No more PR needed for this ticket. 👍

**Pull request merged**